### ISSUE TITLE
Bookie cannot perform Garbage Collection: EntryLogger.scanEntryLog: java.lang.IllegalArgumentException: newCapacity: -610402302 (expected: 0-2147483647)

### ISSUE THREAD

**Issue by hamadodene** [apache/bookkeeper#2264]
**BUG REPORT**

***Describe the bug***

I got this stackstrace while BK was running  (BK version: 4.10.0)
```
20-02-13-16-33-01       Unexpected throwable caught
20-02-13-16-33-01       java.lang.IllegalArgumentException: newCapacity: -610402302 (expected: 0-2147483647)
java.lang.IllegalArgumentException: newCapacity: -610402302 (expected: 0-2147483647)
        at io.netty.buffer.AbstractByteBuf.checkNewCapacity(AbstractByteBuf.java:1419)
        at io.netty.buffer.PooledByteBuf.capacity(PooledByteBuf.java:87)
        at org.apache.bookkeeper.bookie.EntryLogger.scanEntryLog(EntryLogger.java:1014)
        at org.apache.bookkeeper.bookie.EntryLogger.extractEntryLogMetadataByScanning(EntryLogger.java:1132)
        at org.apache.bookkeeper.bookie.EntryLogger.getEntryLogMetadata(EntryLogger.java:1041)
        at org.apache.bookkeeper.bookie.GarbageCollectorThread.extractMetaFromEntryLogs(GarbageCollectorThread.java:557)
        at org.apache.bookkeeper.bookie.GarbageCollectorThread.runWithFlags(GarbageCollectorThread.java:328)
        at org.apache.bookkeeper.bookie.GarbageCollectorThread.safeRun(GarbageCollectorThread.java:309)
        at org.apache.bookkeeper.common.util.SafeRunnable.run(SafeRunnable.java:36)
        at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:515)
        at java.base/java.util.concurrent.FutureTask.runAndReset(FutureTask.java:305)
        at java.base/java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:305)
        at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
        at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
        at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30)
        at java.base/java.lang.Thread.run(Thread.java:830)
```

***To Reproduce***
I'm sorry, I don't have any mean to reproduce it

**Comment by aluccaroni** [comment]
I saw this happening in another cluster. Here are some other logs at the same time:

org.apache.bookkeeper.common.util.SafeRunnable:
```
20-03-26-23-45-17       Unexpected throwable caught
20-03-26-23-45-17       java.lang.IllegalArgumentException: newCapacity: -1890359 (expected: 0-2147483647)
java.lang.IllegalArgumentException: newCapacity: -1890359 (expected: 0-2147483647)
        at io.netty.buffer.AbstractByteBuf.checkNewCapacity(AbstractByteBuf.java:1470)
        at io.netty.buffer.PooledByteBuf.capacity(PooledByteBuf.java:100)
        at org.apache.bookkeeper.bookie.EntryLogger.scanEntryLog(EntryLogger.java:1014)
        at org.apache.bookkeeper.bookie.EntryLogger.extractEntryLogMetadataByScanning(EntryLogger.java:1132)
        at org.apache.bookkeeper.bookie.EntryLogger.getEntryLogMetadata(EntryLogger.java:1041)
        at org.apache.bookkeeper.bookie.GarbageCollectorThread.extractMetaFromEntryLogs(GarbageCollectorThread.java:557)
        at org.apache.bookkeeper.bookie.GarbageCollectorThread.runWithFlags(GarbageCollectorThread.java:328)
        at org.apache.bookkeeper.bookie.GarbageCollectorThread.safeRun(GarbageCollectorThread.java:309)
        at org.apache.bookkeeper.common.util.SafeRunnable.run(SafeRunnable.java:36)
        at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:515)
        at java.base/java.util.concurrent.FutureTask.runAndReset(FutureTask.java:305)
        at java.base/java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:305)
        at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
        at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
        at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30)
        at java.base/java.lang.Thread.run(Thread.java:834)
```

org.apache.bookkeeper.bookie.EntryLogger:
```
20-03-26-20-05-17       Failed to get ledgers map index from: 9.log : Cannot deserialize ledgers map from ledger 158913789951
20-03-26-20-15-17       Failed to get ledgers map index from: 9.log : Cannot deserialize ledgers map from ledger 158913789951
20-03-26-20-25-17       Failed to get ledgers map index from: 9.log : Cannot deserialize ledgers map from ledger 158913789951
20-03-26-20-35-17       Failed to get ledgers map index from: 9.log : Cannot deserialize ledgers map from ledger 158913789951
20-03-26-20-45-17       Failed to get ledgers map index from: 9.log : Cannot deserialize ledgers map from ledger 158913789951
20-03-26-20-55-17       Failed to get ledgers map index from: 9.log : Cannot deserialize ledgers map from ledger 158913789951
20-03-26-21-05-17       Failed to get ledgers map index from: 9.log : Cannot deserialize ledgers map from ledger 158913789951
20-03-26-21-15-17       Failed to get ledgers map index from: 9.log : Cannot deserialize ledgers map from ledger 158913789951
20-03-26-21-25-17       Failed to get ledgers map index from: 9.log : Cannot deserialize ledgers map from ledger 158913789951
20-03-26-21-35-17       Failed to get ledgers map index from: 9.log : Cannot deserialize ledgers map from ledger 158913789951
20-03-26-21-45-17       Failed to get ledgers map index from: 9.log : Cannot deserialize ledgers map from ledger 158913789951
20-03-26-21-55-17       Failed to get ledgers map index from: 9.log : Cannot deserialize ledgers map from ledger 158913789951
20-03-26-22-05-17       Failed to get ledgers map index from: 9.log : Cannot deserialize ledgers map from ledger 158913789951
20-03-26-22-15-17       Failed to get ledgers map index from: 9.log : Cannot deserialize ledgers map from ledger 158913789951
```

org.apache.bookkeeper.bookie.GarbageCollectorThread:
```
20-03-26-15-05-17       Extracting entry log meta from entryLogId: 9
20-03-26-15-15-17       Extracting entry log meta from entryLogId: 9
20-03-26-15-25-17       Extracting entry log meta from entryLogId: 9
20-03-26-15-35-17       Extracting entry log meta from entryLogId: 9
20-03-26-15-45-17       Extracting entry log meta from entryLogId: 9
20-03-26-15-55-17       Extracting entry log meta from entryLogId: 9
20-03-26-16-05-17       Extracting entry log meta from entryLogId: 9
20-03-26-16-15-17       Extracting entry log meta from entryLogId: 9
20-03-26-16-25-17       Extracting entry log meta from entryLogId: 9
20-03-26-16-35-17       Extracting entry log meta from entryLogId: 9
20-03-26-16-45-17       Extracting entry log meta from entryLogId: 9
20-03-26-16-55-17       Extracting entry log meta from entryLogId: 9
20-03-26-17-05-17       Extracting entry log meta from entryLogId: 9
20-03-26-17-15-17       Extracting entry log meta from entryLogId: 9
20-03-26-17-25-17       Extracting entry log meta from entryLogId: 9
20-03-26-17-35-17       Extracting entry log meta from entryLogId: 9
20-03-26-17-45-17       Extracting entry log meta from entryLogId: 9
20-03-26-17-55-17       Extracting entry log meta from entryLogId: 9
20-03-26-18-05-17       Extracting entry log meta from entryLogId: 9
20-03-26-18-15-17       Extracting entry log meta from entryLogId: 9
20-03-26-18-25-17       Extracting entry log meta from entryLogId: 9
20-03-26-18-35-17       Extracting entry log meta from entryLogId: 9
20-03-26-18-45-17       Extracting entry log meta from entryLogId: 9
20-03-26-18-55-17       Extracting entry log meta from entryLogId: 9
20-03-26-19-05-17       Extracting entry log meta from entryLogId: 9
20-03-26-19-15-17       Extracting entry log meta from entryLogId: 9
20-03-26-19-25-17       Extracting entry log meta from entryLogId: 9
20-03-26-19-35-17       Extracting entry log meta from entryLogId: 9
20-03-26-19-45-17       Extracting entry log meta from entryLogId: 9
20-03-26-19-55-17       Extracting entry log meta from entryLogId: 9
20-03-26-20-05-17       Extracting entry log meta from entryLogId: 9
20-03-26-20-15-17       Extracting entry log meta from entryLogId: 9
20-03-26-20-25-17       Extracting entry log meta from entryLogId: 9
```

**Comment by eolivelli** [comment]
We are hitting this problem in several deployments of BK, with different applications (MagNews.com, HerdDB, Majordodo...) since we moved to BK 4.10.0 (perhaps even with 4.9.2).

The fix would be simple, we can consider the file as corrupted and jump outside the loop.

**Comment by eolivelli** [comment]
Even restarting the bookie is not fixing the issue, the file appears corrupted.

is there any change we did that could lead to rewriting/corrupting EntryLogger files ? we are using classic EntryLogger (not multi entry logs).

@jvrao  @sijie @reddycharan @ivankelly  have you ever seen this problem in production ?

**Comment by eolivelli** [comment]
Fast fix for production:
1) look for "Extracting entry log meta from entryLog" line in logs and get the entry log id.
2) Convert the entry log id to base16 and append '.log' (If entry log id is 4053 than look for file fd5.log)
3) remove that file (do a backup!)
4) reboot the bookie

**Comment by sijie** [comment]
This is fixed by #2373

**Comment by kwenzh** [comment]
> Fast fix for production:
> 
> 1. look for "Extracting entry log meta from entryLog" line in logs and get the entry log id.
> 2. Convert the entry log id to base16 and append '.log' (If entry log id is 4053 than look for file fd5.log)
> 3. remove that file (do a backup!)
> 4. reboot the bookie

I have a similar problem in pulsar2.8.1, use org.apache.bookkeeper-bookkeeper-server-4.14.2.jar,    can I fast fix it like this?

![image](https://user-images.githubusercontent.com/38367518/186312473-2d32413f-66a3-438f-bfb1-126016c1c4d6.png)

```
Aug 24, 2022 @ 11:04:32.101 | 03:04:32.101 [GarbageCollectorThread-11-1] INFO  org.apache.bookkeeper.bookie.EntryLogger - Failed to get ledgers map index from: 23624.log : Cannot deserialize ledgers map from ledger 3472891246198208823
Aug 24, 2022 @ 11:04:32.091 | 03:04:32.091 [GarbageCollectorThread-11-1] INFO  org.apache.bookkeeper.bookie.GarbageCollectorThread - Extracting entry log meta from entryLogId: 23624
```
